src/main/java/net/snowflake/client/jdbc/SnowflakeStatementV1.java

```diff
@@ -214,16 +215,18 @@ long executeUpdateInternal(
       sfResultSet.setSession(this.connection.getSFBaseSession());
       updateCount = ResultUtil.calculateUpdateCount(sfResultSet);
       queryID = sfResultSet.getQueryId();
-      statementMetaData =
-          new SFStatementMetaData(
-              sfResultSet.getMetaData(),
-              sfResultSet.getStatementType(),
-              sfResultSet.getNumberOfBinds(),
-              sfResultSet.isArrayBindSupported(),
-              sfResultSet.getMetaDataOfBinds(),
-              true); // valid metadata
+      if (!preparedStatementMetaData.isValidMetaData()) {
```

Comment:
How can we check here, so this object won't create for a Simple Statement (i.e. non prepared statement)?

Reply:
It will be created, but the metadata will not be used. This is similar to how resultsets are created regardless of whether the user needs a resultset object or not. The metadata is not used/hidden for SnowflakeStatement functions.

Reply:
Creating a resultset is a contract as part of any statement execution. But here question is about metadata, which is only make sense for a prepared statement. So we shouldn't create for a simple statement.